node/narwhal/src/helpers/committee.rs

```diff
@@ -324,118 +324,174 @@ pub mod prop_tests {
     use super::*;
 
     use snarkos_account::Account;
+    use std::{
+        collections::HashSet,
+        hash::{Hash, Hasher},
+    };
 
-    use std::{collections::HashSet, hash::Hash};
-
+    use crate::{MAX_COMMITTEE_SIZE, MIN_STAKE};
     use anyhow::Result;
-    use indexmap::IndexMap;
-    use proptest::sample::size_range;
+    use proptest::{
+        collection::{hash_set, SizeRange},
+        prelude::{any, Arbitrary, BoxedStrategy, Just, Strategy},
+        sample::size_range,
+    };
     use rand::SeedableRng;
-    use test_strategy::{proptest, Arbitrary};
+    use test_strategy::proptest;
 
     type CurrentNetwork = snarkvm::prelude::Testnet3;
 
-    #[derive(Arbitrary, Debug, Clone)]
-    pub struct CommitteeInput {
-        #[strategy(0u64..)]
-        pub round: u64,
-        // Using a HashSet here guarantees we'll check the PartialEq implementation on the
-        // `account_seed` and generate unique validators.
-        #[any(size_range(0..32).lift())]
-        pub validators: HashSet<Validator>,
-    }
-
-    #[derive(Arbitrary, Debug, Clone, Eq)]
+    #[derive(Debug, Clone)]
     pub struct Validator {
-        #[strategy(..5_000_000_000u64)]
         pub stake: u64,
-        account_seed: u64,
+        pub account: Account<CurrentNetwork>,
     }
 
-    // Validators can have the same stake but shouldn't have the same account seed.
-    impl PartialEq for Validator {
-        fn eq(&self, other: &Self) -> bool {
-            self.account_seed == other.account_seed
+    impl Arbitrary for Validator {
+        type Parameters = ();
+        type Strategy = BoxedStrategy<Validator>;
+
+        fn arbitrary_with(_: Self::Parameters) -> Self::Strategy {
+            any_valid_validator()
         }
     }
 
-    // Make sure the Hash matches PartialEq.
-    impl Hash for Validator {
-        fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
-            self.account_seed.hash(state);
+    fn to_committee((round, ValidatorSet(validators)): (u64, ValidatorSet)) -> Result<Committee<CurrentNetwork>> {
+        Committee::new(round, validators.iter().map(|v| (v.account.address(), v.stake)).collect())
+    }
+
+    fn validator_set<T: Strategy<Value = Validator>>(
+        element: T,
+        size: impl Into<SizeRange>,
+    ) -> impl Strategy<Value = ValidatorSet> {
+        hash_set(element, size).prop_map(ValidatorSet)
+    }
+
+    fn invalid_round_committee() -> BoxedStrategy<Result<Committee<CurrentNetwork>>> {
+        (Just(0), validator_set(any_valid_validator(), size_range(4..=MAX_COMMITTEE_SIZE as usize)))
```

Comment:
```suggestion
        (Just(0), validator_set(any_valid_validator(), size_range(3..=MAX_COMMITTEE_SIZE as usize)))
```
at least marked with a TODO if it doesn't work yet

Reply:
Actually the original is correct, as we are only generating cases here where the round is invalid (so committee size must be valid).